### ISSUE TITLE
Type inference/conversion issue with sizeof

### ISSUE THREAD

**Issue by Clyybber** [nim-lang/Nim#8980]
```
type
  AnObject = object
    content: int
    otherContent: int

var someObjects = [
  AnObject(content: 1, otherContent: 3),
  AnObject(content: 2, otherContent: 2),
  AnObject(content: 3, otherContent: 1) ]

var someNumbers = [1,2,3]

proc test1() =
  var num: uint64 = (sizeof(someNumbers[0]) * someNumbers.len)
  echo num

#proc test2() =
#  var num: uint64 = (sizeof(someObjects[0]) * someObjects.len) #Also fails with sizeof(AnObject)
#  echo num

test1()

#test2()
```
This works (unexpectedly).
However if you remove the comments, the compilation fails with 
`Error: type mismatch: got <int> but expected 'uint64'

**Comment by Clyybber** [comment]
As pointed out by @TheLemonMan on IRC this happens because sizeof(anobjecttype) returns an int, and is not constant-folded.
So it boils down to sizeof(anobjecttype) not being determinable at compiletime.

**Comment by mratsim** [comment]
You can use `static(sizeof(someNumbers[0])))`.

Probably pending #5664

**Comment by Clyybber** [comment]
Yep #5664 fixes this.

**Comment by Clyybber** [comment]
Closed by #5664
